plugins/woocommerce/client/blocks/assets/js/data/checkout/push-changes.ts

```diff
@@ -83,14 +83,30 @@ const updateCheckoutData = (): void => {
 		localState.doingPush = false;
 		return;
 	}
-
 	// Figure out which additional fields have changed and only send those to the server
 	const changedFields = Object.keys( newCheckoutData.additionalFields )
```

Comment:
Not specific to the changes here but just reading `updateCheckoutData` I noticed that if validation fails [we set the local cache, to be the current state](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/blocks/assets/js/data/checkout/push-changes.ts#L107-L111). However, [if the check on `activePaymentMethod` fails, then we don't](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/client/blocks/assets/js/data/checkout/push-changes.ts#L82-L85) update the local cache at all. Just curious as to why?

Reply:
Sadly I'm not sure, this was written by Alex, my best guess is that selecting payment failing is shouldn't update the fields cache, as they would trigger from different paths, I will digg more into it later on.